handlers/ticket-tailor-webhook/src/validateRequest.ts

```
@@ -0,0 +1,99 @@
+		buyer_details: {
+			email: string;
+		};
+	};
+}
+
+export const getTimestampAndSignature = (
+	record: SQSRecord,
+): [string, string] => {
+	const signatureWithTs =
+		record.messageAttributes['tickettailor-webhook-signature']?.stringValue;
+	if (typeof signatureWithTs === 'string') {
+		const timestamp = signatureWithTs.split(',')[0]?.split('=')[1];
+		const signature = signatureWithTs.split(',')[1]?.split('=')[1];
+		if (timestamp && signature) {
+			if (!isNaN(Number(timestamp))) {
+				return [timestamp, signature];
+			} else {
+				throw new Error(
+					`Invalid value for MessageAttribute 'tickettailor-webhook-signature' -> timestamp: ${timestamp}. Timestamp should be a numeric string.`,
+				);
+			}
+		} else {
+			throw new Error(
+				`Invalid formatting of MessageAttribute 'tickettailor-webhook-signature': ${signatureWithTs}. Missing timestamp or signature.`,
+			);
+		}
+	} else {
+		throw new Error(
+			'No valid value found for MessgeAttritbute: tickettailor-webhook-signature on incoming request.',
+		);
+	}
+};
+
+export const isWithinTimeWindow = (
+	allowedTimeWindowInSeconds: number,
+	timestamp: string,
+	currentDateTime: Date,
+): boolean => {
+	const timestampEpochSeconds = Number(timestamp);
+	const timeDiff = currentDateTime.getSeconds() - timestampEpochSeconds;
+	if (timeDiff < 0) {
+		throw new Error(
+			`Invalid Webhook Signature: timeStamp ${timestamp} is later than current time. Check it is not using EpochMillis.`,
+		);
+	} else {
+		return timeDiff <= allowedTimeWindowInSeconds;
+	}
+};
+
+export const hasMatchingSignature = (
+	timestamp: string,
+	signature: string,
+	record: SQSRecord,
+	validationSecret: HmacKey,
+): boolean => {
+	const hash = createHmac('sha256', validationSecret.secret)
+		.update(timestamp.concat(record.body))
+		.digest('hex');
+	return timingSafeEqual(Buffer.from(hash), Buffer.from(signature));
+};
+
+export const validateRequest = (
+	record: SQSRecord,
+	validationSecret: HmacKey,
+): void => {
```

This method throws errors in place rather than returning a boolean value for success or failure. Reason for this is that there are too many possible ways to fail the request validation now. Simply passing a boolean back would lose too much information.